plugins/admin_console/services/EntryAdminService.php

```diff
@@ -81,4 +93,113 @@ function getTracksAction($entryId)
 		$response->totalCount = count($dbList);
 		return $response;
 	}
+
+	/**
+	 * Restore deleted entry.
+	 *
+	 * @action restoreDeletedEntry
+	 * @param string $entryId
+	 * @return KalturaBaseEntry The restored entry
+	 */
+	public function restoreDeletedEntryAction($entryId)
+	{
+		$deletedEntry = entryPeer::retrieveByPKNoFilter($entryId);
+		if (!$deletedEntry)
+			throw new KalturaAPIException(KalturaErrors::ENTRY_ID_NOT_FOUND, $entryId);
+
+		$fileSyncKeys = array();
+		foreach (self::$fileSyncKeysToRestore as $key) {
+			$fileSyncKeys[] = $deletedEntry->getSyncKey($key);
+		}
+
+		$c = new Criteria();
+		$c->add(assetPeer::ENTRY_ID, $entryId, Criteria::EQUAL);
+		assetPeer::setUseCriteriaFilter(false);
+		$deletedAssets = assetPeer::doSelect($c);
+		assetPeer::setUseCriteriaFilter(true);
+
+		foreach($deletedAssets as $deletedAsset)
+		{
+			array_push($fileSyncKeys, $deletedAsset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_ASSET), $deletedAsset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_CONVERT_LOG));
+		}
+
+		$fileSyncs = array();
+		FileSyncPeer::setUseCriteriaFilter(false);
+		foreach ($fileSyncKeys as $fileSyncKey)
+		{
+			$fileSyncs = array_merge($fileSyncs, FileSyncPeer::retrieveAllByFileSyncKey($fileSyncKey));
+		}
+		FileSyncPeer::setUseCriteriaFilter(true);
+
+		if (!$this->validateEntryForRestoreDelete($deletedEntry, $fileSyncs, $deletedAssets))
+			throw new KalturaAPIException(KalturaAdminConsoleErrors::ENTRY_ASSETS_WRONG_STATUS_FOR_RESTORE, $entryId);
+
+		$this->restoreFileSyncs($fileSyncs);
+
+		//restore assets
+		foreach($deletedAssets as $deletedAsset)
+		{
+			$deletedAsset->setStatus(asset::ASSET_STATUS_READY);
+			$deletedAsset->save();
+		}
+
+		//restore entry
+		$deletedEntry->setStatusReady();
+		$deletedEntry->setThumbnail($deletedEntry->getFromCustomData("deleted_original_thumb"), true);
+		$deletedEntry->setData($deletedEntry->getFromCustomData("deleted_original_data"),true); //data should be resotred even if it's NULL
+		$deletedEntry->save();
+
+		kEventsManager::flushEvents();
+		kMemoryManager::clearMemory();
+
+		$entry = KalturaEntryFactory::getInstanceByType($deletedEntry->getType(), true);
+		$entry->fromObject($deletedEntry, $this->getResponseProfile());
+		return $entry;
+	}
+
+	protected function restoreFileSyncs( array $fileSyncs )
+	{
+		foreach ($fileSyncs as $fileSync)
+		{
+			$shouldUnDelete = false;
+			if (($fileSync->getFileType() == FileSync::FILE_SYNC_FILE_TYPE_FILE && file_exists($fileSync->getFullPath()))
```

Comment:
File-sync objects for other DCs will never return true for file_exists.
If you want to restore them too, you can call another API internally to the other DC.